Add support for `depth` option in wrap-git

fiesh opened the pull request:
This allows cloning subprojects shallowly.  It works recursively for a
subproject's submodules in case `clone-recursive` is set to `true`.

ignatenkobrain commented on mesonbuild/wrap/wrap.py:
```diff
@@ -226,16 +226,21 @@ def get_file(self):
+        if self.wrap.values.get('depth', '') != '':
+            git_options.extend(['--depth', self.wrap.values.get('depth')])
+        subprocess.check_call(['git', 'fetch'] + git_options + ['origin', revno],
+                cwd=self.dirname)
```
__[Flake8]__ [E128] continuation line under-indented for visual indent ([link](https://sider.review/gh/repos/19784232/pulls/5932#issue-4580791))

<sub>posted by Sider</sub>

fiesh replied:
How do I know what the right level of indentation would be?

ignatenkobrain commented on mesonbuild/wrap/wrap.py:
```diff
@@ -226,16 +226,21 @@ def get_file(self):
+        # so we follow https://stackoverflow.com/a/43136160
+        subprocess.check_call(['git', 'init', self.directory], cwd=self.subdir_root)
+        subprocess.check_call(['git', 'remote', 'add', 'origin', self.wrap.get('url')],
+                               cwd=self.dirname)
```
__[Flake8]__ [E127] continuation line over-indented for visual indent ([link](https://sider.review/gh/repos/19784232/pulls/5932#issue-4580790))

<sub>posted by Sider</sub>

QuLogic replied:
This should be on the same column as the `[`; it is an argument to the function and thus related to its opening parenthesis.

fiesh replied:
Thank you.  I fixed the indentation and `fixup`ed the last commit.

fiesh commented:
Ok Flake8 tells me something is under- and over-indented, but instead of suggesting a fix it supplies a dead link :-(

scivision commented:
I like `autopep8`, it virtually always does the right thing, at least for what I've used it for.
You might try something like:

```python
autopep8 --select E127,E128 -i mesonbuild/wrap/wrap.py
```

jpakkane commented:
Looks nice, just needs to fix the version number to 0.52.0 and, since this is new functionality, a release note snippet needs to be added.

fiesh commented:
> Looks nice, just needs to fix the version number to 0.52.0 and, since this is new functionality, a release note snippet needs to be added.

I fixed these two issues and rebased the commit.  Please take another look.

Thank you!

fiesh commented:
Rebased and resolved the "conflict" in the release notes.

fiesh commented:
Is this a sporadic timeout error in Travis CI?  I don't understand what the error is.

fiesh commented:
Seems very much like a sporadic error, but I have no idea how to restart the build job.

jpakkane commented:
:crossed_swords: conflicts. Also, you should never merge master to your branch, but instead keep the MR up to date by rebasing [as explained in the documentation](https://mesonbuild.com/Contributing.html#strategy-for-merging-pull-requests-to-trunk).

fiesh commented:
> crossed_swords conflicts. Also, you should never merge master to your branch, but instead keep the MR up to date by rebasing [as explained in the documentation](https://mesonbuild.com/Contributing.html#keeping-pull-requests-up-to-date).

Ah man I used the new github fancy features and thought they for sure will rebase...  I'll do it by hand

jpakkane commented:
The changes should go in a snippet file, not in the release note file. I have fixed master so that the changes that were (erroneously) there have been moved to standalone files. Please do the same here.

fiesh commented:
> The changes should go in a snippet file, not in the release note file. I have fixed master so that the changes that were (erroneously) there have been moved to standalone files. Please do the same here.

Did that.  From what I could tell, the snippet file is not included anywhere, do knowing from which version on the snippet took effect is done via git log?

jpakkane commented:
Release notes are generated [with this script](https://github.com/mesonbuild/meson/blob/master/docs/genrelnotes.py).
